Assignment1/ho-taek_BESOPTAssignment/app/src/main/java/com/htk/sopt1/SignInActivity.kt

```diff
@@ -0,0 +1,101 @@
+package com.htk.sopt1
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat.startActivityForResult
+import com.htk.sopt1.HomeActivity
+import com.htk.sopt1.databinding.ActivitySigninBinding
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySigninBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ///LayoutInflater는 xml을 activity에 뿌려주는 다리와 같은 역할을 함.
+        binding = ActivitySigninBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+
+        initButtonClickEvent()
+        signUpButtonClickEvent()
+    }
+
+    private fun initButtonClickEvent() {
+        binding.btLogin.setOnClickListener {
+            val userId = binding.etId.text.toString()
+            val userPassword = binding.etPassword.text.toString()
+            // 값이 Null이거나 비어있을 경우 괄호 부분을 실행
+            if (userId.isNullOrBlank() or userPassword.isNullOrBlank()) {
```

Comment:
```suggestion
        private fun isLoginInputValuesEmpty(id: String, password: String) = id.isNullOrBlank() or password.isNullOrBlank()

        if (isLoginInputValuesEmpty(userId, userPassword)) {
            etc
        }
```

이 부분은 다른 친구들한테도 말을 한건데, 이 부분을 함수로 위와 같이 함수화를 시켰을 때의 이점을 한 번 생각해보면 좋을 것 같아요.

Reply:
제 생각에는 코드가 간결해져서 가독성이 높아집니다

Reply:
이외에도 다른 장점은 또 어떤게 있을까요? 만약에 로그인을 할 때 다른 입력창이 추가로 필요하다면? 기존 로직에 비해 수정된 로직에서는 어떤 장점을 보여줄 수 있을까요?

Reply:
기존 로직에서는 다른 입력창에서 또 써줘야하지만 함수화 하였을때는 함수를 가져다 쓰기만 하면 되기 때문에 불필요하게 코드를 작성할 필요가 없다는 장점 또한 있습니다!